server/http_endpoints_test.go

```diff
@@ -263,6 +263,7 @@ func TestEndpoints_UpstreamBasicAuthAndXFF(t *testing.T) {
 func TestEndpoints_OAuth2(t *testing.T) {
```

Comment:
As mentioned below some cases are missing. Replacing an existing test, which seems to test the default behaviour, is not helpful. Add additional cases to test the default, disabled and >default .

Reply:
Done